planet/api/http.py

```diff
@@ -13,135 +13,176 @@
 # the License.
 
 """Functionality to perform HTTP requests"""
-
-# import http.client as http_client
+import asyncio
 import logging
-import os
-import re
-import time
 
-from requests import Session
-from requests.compat import urlparse
+import httpx
 
 from . import exceptions, models
 from . __version__ import __version__
 
-
-LOGGER = logging.getLogger(__name__)
-
-DEFAULT_CHUNK_SIZE = 32 * 1024
-
-USE_STRICT_SSL = not (os.getenv('DISABLE_STRICT_SSL', '').lower() == 'true')
-
+RETRY_COUNT = 5
 RETRY_WAIT_TIME = 1  # seconds
 
+LOGGER = logging.getLogger(__name__)
 
-# def setup_logging():
-#     log_level = LOGGER.getEffectiveLevel()
-#     urllib3_logger = logging.getLogger(
-#         'requests.packages.urllib3')
-#     urllib3_logger.setLevel(log_level)
-#
-#     # if debug level set then its nice to see the headers of the request
-#     if log_level == logging.DEBUG:
-#         http_client.HTTPConnection.set_debuglevel(1)
-#     else:
-#         http_client.HTTPConnection.set_debuglevel(0)
-#
 
-def _log_request(req):
-    LOGGER.info('%s %s %s %s', req.method, req.url, req.params, req.data)
+class Session():
+    '''Context manager for asynchronous communication with the Planet server.
 
+    Authentication for Planet servers is given as ('<api key>', '').
```

Comment:
Did we want this to reference something a little more abstract like `planet.get_auth()`?

Reply:
the authentication work is captured in #248 - this will be changed there.